__factories__/clients/firebase.js

```diff
@@ -0,0 +1,31 @@
+import {Factory} from 'rosie';
+
+class FirebaseError extends Error {
+  constructor(args) {
+    super(args.name);
+    this.code = args.name;
+    this.credential = args.credential;
+    Error.captureStackTrace(this, FirebaseError);
+  }
+}
+
+export const credential = new Factory().attrs({
+  providerId: 'github.com',
+  accessToken: 'abc123',
+});
+
+export const githubProfile = new Factory().attrs({
+  login: 'popcoder',
+});
```

Comment:
The top-level principle for organizing factories should be where the data they’re generating would come from in the real application—in this case this profile object would come from the GitHub client, so this should live in `__factories__/clients/github.js`.

Also, as I mentioned before, we’ll want to generate factory data that is a reasonably complete representation of what the code we’re testing will see. I don’t necessarily think we need to capture _every_ property of large objects like a GitHub profile, but we should try to capture a subset that generously covers all the properties Popcode might care about (this is a bit subjective, but I trust your judgement). As for where to find that information, [API docs](https://developer.github.com/v3/users/#response-with-public-and-private-profile-information) are probably your best bet; throwing a `debugger` or a `console.log` in the application code and then capturing a real example is another fine approach.

Reply:
Looks like this is still outstanding?

Reply:
Huh, I thought I changed this already. I'll check it out again.